src/ThreeMFParser.cpp

```diff
@@ -127,7 +132,22 @@ std::string ThreeMFParser::sceneToString(Scene scene)
         item.append_attribute("objectid") = scene_node->getId().c_str();
         item.append_attribute("transform") = scene_node->getTransformation().c_str();
     }
-
+    
+    for(const std::pair<std::string, MetadataEntry>& metadata_pair: scene.getMetadata())
+    {
+        pugi::xml_node metadata_node = model_node.append_child("metadata");
+        metadata_node.append_attribute("name") = metadata_pair.first.c_str();
+        metadata_node.text().set(metadata_pair.second.value.c_str());
```

Comment:
These 9 lines (140-148) are a repeat of the ones +- 50 lines above (88-96).

Consider making this into a function (but I don't think it's blocking for component test or acceptance, just a pet-peeve).

Reply:
I considered this when I wrote the code twice. It can be done by making a function that takes an XML node by reference and a map of settings, and serialises those settings into the XML node. Not sure if that really makes the code easier to read though, because that's not a simple to understand operation. So I've left it like this
I'd say we should consider it again if the duplicated code needs to be modified often and we find ourselves doing the same work multiple times frequently because of this.

Reply:
Fair enough.